src/pages/explorePage/ExplorePage.tsx

```diff
@@ -99,6 +138,41 @@ const ExplorePageContent = ({
     hasPrereqs,
   };
 
+  const mapFilterStateToURL = (fs: SearchFilterState) => {
+    return {
+      exclude: fs.courseCodes
+        .map((bool, index) => (bool ? null : index))
+        .filter((index) => index !== null),
+      numCourseRatings: !fs.numCourseRatings ? null : fs.numCourseRatings,
+      numProfRatings: !fs.numProfRatings ? null : fs.numProfRatings,
+      courseTaught: !fs.courseTaught ? null : fs.courseTaught,
+      currentTerm: !fs.currentTerm ? null : fs.currentTerm,
+      nextTerm: !fs.nextTerm ? null : fs.nextTerm,
+      hasNoPrereqs: fs.hasPrereqs ? null : true,
+    };
+  };
+
+  useEffect(() => {
+    window.history.pushState(
```

Comment:
Oh nice this works because we're not using React Router so there's no re-render?

Reply:
Yeah, I think the reason it's not infinite looping is because we're using the `window` object , which you mentioned is seperate.